Userland/Applications/PixelPaint/ImageEditor.cpp

```diff
@@ -96,6 +96,78 @@ void ImageEditor::paint_event(GUI::PaintEvent& event)
         }
     }
 
+    if (m_show_rulers) {
+        const int ruler_thickness = 20;
+        const auto ruler_bg_color = palette().color(Gfx::ColorRole::Tray);
+        const auto ruler_fg_color = Color::LightGray;
+
+        // Ruler background
+        painter.fill_rect({ { 0, 0 }, { ruler_thickness, rect().height() } }, ruler_bg_color);
+        painter.fill_rect({ { 0, 0 }, { rect().width(), ruler_thickness } }, ruler_bg_color);
+
+        const int ruler_step = [&]() {
+            const auto step_target = 80 / m_scale;
+            const auto max_factor = 5;
+            for (int factor = 0; factor < max_factor; ++factor) {
+                if (step_target <= 1 * (float)pow(10, factor))
+                    return 1 * pow(10, factor);
+                if (step_target <= 2 * (float)pow(10, factor))
+                    return 2 * pow(10, factor);
+                if (step_target <= 5 * (float)pow(10, factor))
+                    return 5 * pow(10, factor);
+            }
+            return 1 * pow(10, max_factor);
+        }();
```

Comment:
What's the point of creating a lambda here if you're just calling it instantly and not even storing it to use it later? Why not just have the code in line?

You also seem to be capturing everything by reference and just using one values, which isn't ideal either.

Reply:
Using a lambda here makes the logic flow somewhat simpler IMO since the return statements break the for loop and the last return sets a the edge case. But sure, the same logic can be expressed inline as well. :)

Woops, I'll fix that capture.

Reply:
maybe just make this a private method if you think the early returns help. The only captured variable here should be `m_scale` which would be accessible by any class methods too.